zarr/storage.py

```diff
@@ -781,12 +781,15 @@ def __setitem__(self, key, value):
 
     def __delitem__(self, key):
         path = os.path.join(self.path, key)
+        temp_path = path + '.partial'
         if os.path.isfile(path):
-            os.remove(path)
+            os.rename(path, temp_path)
```

Comment:
This rename will fail if the `.partial` file already exist.

Reply:
Actually that's only true on Windows. Unix it is deleted silently. Please see [this note in the docs]( https://docs.python.org/3/library/os.html#os.rename ).

That said, I think this is an unlike scenario and even if it occurred would expect we want it to fail (see below).

We only create a `*.partial` file currently if we are writing a value, but have not completed writing yet. Also we make sure to delete the corresponding file `*` before creating `*.partial`. So both should not exist at the same time. If we do ever see a `*.partial` file, that's an indication of an incomplete write (IOW data corruption), in which case we should fail somehow. Admittedly we may want to adjust the failure to make it more informative and less reliant on internal spec.

So agree that behavior should be adjusted, but think it should be in the direction of always failing if a `*.partial` file exists.